backend/src/plugins/authorization/authorization.plugin.ts

```diff
@@ -25,11 +27,16 @@ const auth: FastifyPluginAsync<Options> = async (fastify, opts) => {
 
   fastify.addHook(ControllerHook.ON_REQUEST, async (request, reply) => {
     try {
-      const isWhiteRoute = opts.routesWhiteList.some(
-        (route) => route === request.routerPath,
+      const whiteRoute = opts.routesWhiteList.find(
+        (route) => route.route === request.routerPath,
       );
 
-      if (isWhiteRoute) {
+      const isWhiteRoute = Boolean(whiteRoute);
+
+      if (
+        isWhiteRoute &&
+        whiteRoute?.methods.includes(request.method as HttpMethod)
+      ) {
```

Comment:
```suggestion
      const isWhiteRoute = Boolean(whiteRoute);
      
      if (!isWhiteRoute) {
        return
      }
      
      const isAllowedMethod = (whiteRoute as WhiteRoute).methods.includes(request.method as HttpMethod);

      if (!isAllowedMethod) {
        return
```

Reply:
The logic is actually not correct here. We can't return if `!isWhiteRoute` or if `!isAllowedMethod`, because for these cases we have to attach the user to `request`